### PR TITLE
proxy-config documentation

### PR DESCRIPTION
**Pull request by leon**
documentation on how to use `--proxy-config`

### CODE REVIEW

**Comment by leon** [comment]
I signed it!

**Comment by mmrath** [comment]
I think it is worth mentioning that proxy confguration would support all the configuration mentioned at https://webpack.github.io/docs/webpack-dev-server.html#proxy except support for functions in configuration.

Good work @leon

**Comment by leon** [comment]
@mmrath I've added the link :)

**Comment by jasonswett** [comment]
The config file in these instructions actually didn't work for me. What did work for me was something slightly different:

```json
{
  "/api": {
    "target": "http://localhost:3000",
    "secure": "false"
  }
}
```

(`/api` instead of `api/*`)

**Comment by leon** [comment]
@jasonswett We can change it to `/api` it works for me too, and is easier to understand.

**Comment by gonzofish** [comment]
Thanks for getting this doc out there! Helped me out big time.

**Comment by filipesilva** [comment]
Thanks for documenting this feature, we'll merge this soon!

File: README.md [hunk 1]
```diff
@@ -186,6 +187,33 @@ Before running the tests make sure you are serving the app via `ng serve`.
 
 End-to-end tests are run via [Protractor](https://angular.github.io/protractor/).
 
+### Proxy To Backend
+Using the proxying support in webpack's dev server we can highjack certain urls and send them to a backend server.
+We do this by passing a file to `--proxy-config`
+
+Say we have a server running on `http://localhost:3000/api` and we want all calls th `http://localhost:4200/api` to go to that server.
+
```
Comment (mrgoos) [inline README.md]: `th` = `to`.

Thanks for the documentation!

**Comment by imalhasaranga** [comment]
i'm getting the following error while debugging

`[HPM] PROXY ERROR: CERT_HAS_EXPIRED. localhost -> https://api.mydomain.com`

this is our dev server so, is there a way I can add following to get rid of the problem or is there a better way of fixing this ?

`process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";`

**Comment by gus4no** [comment]
Any idea on how to have different targets per environment?

**Comment by gonzofish** [comment]
@gus4no you could have different NPM tasks which call `ng serve --proxy proxies/dev.config.json` in dev or `ng serve --proxy proxies/integration.config.json` or whatever

not an automated solution, but something...

**Comment by maximelafarie** [comment]
Always nothing about --proxy option in official documentation ?

**Comment by filipesilva** [comment]
@maximelafarie it's in the wiki now: https://github.com/angular/angular-cli/wiki/stories-proxy

**Comment by stockmind** [comment]
It could seems silly but a note/remind like "After every edit to proxy.config.json a restart of npm process is needed to apply them" could save some time in late hour working! :smile:

**Pull request closed without merging**